packages/engine-backend/router/resourceRouter.ts

```diff
@@ -22,6 +22,52 @@ export {type inferProcedureInput} from '@openint/trpc'
 
 const tags = ['Core']
 
+async function performResourceCheck(
```

Comment:
Does the checkResource endpoint itself also use this function?

Reply:
yes, is this supposed to refresh tokens? it doesn't seem to do so and hence i wanted to your guidance / to take it over

Reply:
is it working as it is written here right now?

Reply:
No, its not working.